module/support/logging/FileLogHandler/src/FileLogHandler.cpp

```diff
@@ -107,16 +107,36 @@ namespace module::support::logging {
 
             // Output the level
             switch (message.level) {
-                case NUClear::TRACE: logFile << source << "TRACE: "; break;
-                case NUClear::DEBUG: logFile << source << Colour::green << "DEBUG: "; break;
-                case NUClear::INFO: logFile << source << Colour::brightblue << "INFO: "; break;
-                case NUClear::WARN: logFile << source << Colour::yellow << "WARN: "; break;
-                case NUClear::ERROR: logFile << source << Colour::brightred << "ERROR: "; break;
-                case NUClear::FATAL: logFile << source << Colour::brightred << "FATAL: "; break;
+                case NUClear::TRACE: log_file << source << "TRACE: "; break;
+                case NUClear::DEBUG: log_file << source << Colour::green << "DEBUG: "; break;
+                case NUClear::INFO: log_file << source << Colour::brightblue << "INFO: "; break;
+                case NUClear::WARN: log_file << source << Colour::yellow << "WARN: "; break;
+                case NUClear::ERROR: log_file << source << Colour::brightred << "ERROR: "; break;
+                case NUClear::FATAL: log_file << source << Colour::brightred << "FATAL: "; break;
             }
 
             // Output the message
-            logFile << message.message << std::endl;
+            log_file << message.message << std::endl;
+        });
+
+
+        // This checks that we haven't reached the max_size
+        log_check_handler = on<Every<5, std::chrono::seconds>, Sync<FileLogHandler>, Single>().then([this]() {
+            long size = 0;
+            for (auto& f : std::filesystem::recursive_directory_iterator(log_file_name.remove_filename())) {
+                if (f.is_regular_file()) {
+                    size += f.file_size();
+                }
+            }
+            if (size >= max_size) {
+                logging_reaction.unbind();
+                stats_reaction.unbind();
+                log_check_handler.unbind();
```

Comment:
Disabling is probably fine here
```suggestion
                logging_reaction.disable();
                stats_reaction.disable();
                log_check_handler.disable();
```